pathetic-model/src/main/java/org/patheloper/model/pathing/pathfinder/AStarPathfinder.java

```diff
@@ -62,27 +65,51 @@ private void evaluateNewNodes(
       Set<PathPosition> examinedPositions,
       Node currentNode,
       List<PathFilter> filters,
+      List<PathFilterStage> filterStages,
       boolean allowingDiagonal) {
+
     Collection<Node> newNodes =
-        fetchValidNeighbours(examinedPositions, currentNode, filters, allowingDiagonal);
+        fetchValidNeighbours(
+            examinedPositions, currentNode, filters, filterStages, allowingDiagonal);
+
     for (Node newNode : newNodes) {
-      nodeQueue.insert(newNode.getHeuristic().get(), newNode);
+      double priorityAdjustment = calculatePriorityAdjustment(newNode, filterStages);
+      double adjustedCost = newNode.getHeuristic().get() - priorityAdjustment;
+      nodeQueue.insert(adjustedCost, newNode);
+    }
+  }
+
+  private double calculatePriorityAdjustment(
+      Node node, List<PathFilterStage> filterStages) {
+    if (!pathfinderConfiguration.isPrioritizing()) return 0.0;
```

Comment:
Does this not mean that if we are not prioritizing then it will just not test the filter stages?

I assumed that if we added FilterStages we would be removing just standalone filters and that you would just have single filter stages>

Reply:
No, this simply means, if prioritizing is not active, it will simply give no priority adjustment to the node.